pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -71,6 +80,10 @@ func (p *Probe) checkConfigStatus() error {
 
 // isEnvoyReady checks to ensure that Envoy is in the LIVE state and workers have started.
 func (p *Probe) isEnvoyReady() error {
+	// If proxy is terminating, just return the draining state.
+	if p.proxyTerminating {
```

Comment:
Instead of a goroutine setting this and a context, can we just check if the context is done here?

Reply:
I think that is a good idea

Reply:
I tried but could not - but moved to atomic bool to address the concurrency issue

Reply:
Why not? If the issue is blocking on Done() you can use `select` with `default`